pie-metal/src/pie_metal/_internal/mps_rope.py

```
@@ -0,0 +1,212 @@
+        # Verify tensors are on MPS device
+        if input_qk.device.type != 'mps':
+            raise RuntimeError(f"input_qk must be on MPS device, got {input_qk.device}")
+        if position_ids.device.type != 'mps':
+            position_ids = position_ids.to('mps')
+
+        # Get dimensions
+        num_tokens, num_heads, head_size = input_qk.shape
+
+        # DEBUG: Log input parameters
+        if DEBUG_ENABLED and DEBUG_VERBOSITY >= VERBOSITY_DETAILED:
+            print(f"\n🔍 [RoPE MPS] Input parameters:")
```

Ruff (0.13.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)